src/main/java/me/kodysimpson/simpapi/utility/TimeUtil.java

```diff
@@ -0,0 +1,181 @@
+
+
+        String day;
+        if (shortened)
+            day = "d";
+        else
+            day = (toDays == 1 ? "day" : "days");
+
+        String hour;
+        if (shortened)
+            hour = "h";
+        else
+            hour = (toHour == 1 ? "hour" : "hours");
+
+        String minute;
+        if (shortened)
+            minute = "m";
+        else
+            minute = (toMin == 1 ? "minute" : "minutes");
+
+        String second;
+        if (shortened)
+            second = "s";
+        else
+            second = (toSec == 1 ? "second" : "seconds");
+
+        return (dayNotZero ? toDays + (shortened ? "" : " ") + day + " " : "") +
+                (hourNotZero ? toHour + (shortened ? "" : " ") + hour + " " : "") + (minuteNotZero ? toMin + (shortened ? "" : " ")
+                + minute + " " : "") + (secondNotZero ? toSec + (shortened ? "" : " ") + second : "");
+    }
+
+    public static Long longFromString(String s) {
+        StringBuilder builder = new StringBuilder();
+        int seconds = 0;
+        int minutes = 0;
+        int hours = 0;
+        int days = 0;
+        int weeks = 0;
+        for (char c : s.toCharArray()) {
+            if (Character.isDigit(c)) {
+                builder.append(c);
+            } else {
+                switch (c) {
+                    case 's':
+                        if (builder.length() != 0) {
+                            seconds += Integer.parseInt(builder.toString());
+                            builder = new StringBuilder();
+                        }
+                        break;
+                    case 'm':
+                        if (builder.length() != 0) {
+                            minutes += Integer.parseInt(builder.toString());
+                            builder = new StringBuilder();
+                        }
+                        break;
+                    case 'h':
+                        if (builder.length() != 0) {
+                            hours += Integer.parseInt(builder.toString());
+                            builder = new StringBuilder();
+                        }
+                        break;
+                    case 'd':
+                        if (builder.length() != 0) {
+                            days += Integer.parseInt(builder.toString());
+                            builder = new StringBuilder();
+                        }
+                        break;
+                    case 'w':
+                        if (builder.length() != 0) {
+                            weeks += Integer.parseInt(builder.toString());
+                            builder = new StringBuilder();
+                        }
+                        break;
+                    default:
+                        throw new IllegalArgumentException("Not a valid duration format.");
+                }
+            }
+        }
+        return 1000L * (seconds + minutes * 60L + hours * 60 * 60L + days * 24 * 60 * 60L + weeks * 7 * 24 * 60 * 60L);
```

Comment:
TimeUnit can do this math for you, I highly recommend you use that instead of this.